plugins/lua/buildingplan/unlink_mechanisms.lua

```diff
@@ -0,0 +1,678 @@
+    {
+        widgets.Label
+        {
+            frame = {t=0, l=0, r=0},
+            text = self.message,
+        },
+        widgets.HotkeyLabel
+        {
+            frame = {b=2, l=0},
+            label = "Yes, proceed",
+            key = "SELECT",
+            auto_width = true,
+            on_activate = self:callback("proceed"),
+        },
+        widgets.HotkeyLabel
+        {
+            frame = {b=0, l=0},
+            label = "Suppress this warning until DF restarts",
+            key = "CUSTOM_SHIFT_S",
+            auto_width = true,
+            on_activate = self:callback("suppress"),
+        },
+    }
+end
+
+function ConfirmWindow:proceed()
+    self.parent_view:dismiss()
+    self.propagate_fn()
+end
+
+function ConfirmWindow:suppress()
+    self.parent_view:dismiss()
+    self.propagate_fn(true)
+end
+
+ConfirmScreen = defclass(ConfirmScreen, gui.ZScreenModal)
+ConfirmScreen.ATTRS
+{
+    focus_path = "mech/confirm",
+    title = DEFAULT_NIL,
+    message = DEFAULT_NIL,
+    desired_lines = 1,
+    propagate_fn = DEFAULT_NIL,
+}
+
+function ConfirmScreen:init()
+    self:addviews
+    {
+        ConfirmWindow
+        {
+            frame_title = self.title,
+            message = self.message,
+            desired_lines = self.desired_lines,
+            propagate_fn = self.propagate_fn,
+        }
+    }
+end
+
+-- ----------------------
+-- MechLinkOverlay
+--
+
+MechLinkOverlay = defclass(MechLinkOverlay, overlay.OverlayWidget)
+MechLinkOverlay.ATTRS
+{
+    desc = "Allows unlinking mechanisms from buildings.",
+    default_enabled = true,
+    overlay_only = true,
+    frame = {b=4, r=40, w=56, h=26},
+    viewscreens = {},
+}
+
+for _,v in ipairs(valid_build) do
+    table.insert(MechLinkOverlay.ATTRS.viewscreens, "dwarfmode/ViewSheets/BUILDING/"..v.."/LinkedBuildings")
+end
+
+function MechLinkOverlay:init()
+    self.num_buttons = 0
+    self.links = {}
+
+    self:addviews
+    {
+        widgets.CycleHotkeyLabel
+        {
+            frame = {b=0, l=0, w=36},
+            label = "Unlink mode",
+            key = "CUSTOM_M",
+            options =
+            {
+                {value=0, label="Don't auto free", pen=COLOR_DARKGRAY},
+                {value=1, label="Free local mechanism", pen=COLOR_GREEN},
+                {value=2, label="Free target mechanism", pen=COLOR_GREEN},
+                {value=3, label="Free local and target", pen=COLOR_GREEN},
+            },
+            initial_option = saved_mode,
+            on_change = function(val) saved_mode = val end,
+        },
+        widgets.HotkeyLabel
+        {
+            frame = {b=0, r=3},
+            key = "CUSTOM_SHIFT_U",
+            label = "Unlink all",
+            auto_width = true,
+            on_activate = self:callback("ask_unlink_all"),
+            enabled = function() return next(self.links) end,
+        },
+        widgets.Scrollbar --Work around for https://dwarffortressbugtracker.com/view.php?id=12721
```

Comment:
There's no indication this is added by DFHack.

Reply:
An added scrollbar isn't too worrisome. I added a similar scrollbar to the embark unit list for when `startdwarf` gives you more dwarves than will fit on the screen.